contrib/gce/sgc

```diff
@@ -0,0 +1,263 @@
+mv $scratch/.git /home/ubuntu
+rm -rf $scratch
+cd /home/ubuntu
+git checkout -- .
+cd /root
+# clone solvcon.
+git clone https://github.com/solvcon/solvcon /var/opt/solvcon
+# finalize.
+chown -R ubuntu.ubuntu /home/ubuntu
+""" # provision_basic
+
+        shutdown_cmd="""# clean up and shutdown.
+rm -rf /home/*/.ssh
+shutdown -h now
+""" # shutdown_cmd
+
+        homedisk_cmd="""# mount workspace
+mount -o discard,defaults /dev/sdb /home
+""" # homedisk
+
+        allow_password_cmd="""# enable ssh password login
+sed -i 's/PasswordAuthentication no/PasswordAuthentication yes/' /etc/ssh/sshd_config
+/etc/init.d/ssh restart
+""" # allow_password
+
+        ret = ""
+        if provision:
+            ret += provision_cmd
+            if shutdown:
+                ret += shutdown_cmd
+        elif homedisk:
+            ret += homedisk_cmd
+        if allow_password:
+            ret += allow_password_cmd
+        return ret
+
+    def __call__(self, args):
+        if args.homedisk and args.provision:
+            self.parser.error('option --homedisk conflicts to --provision')
+        if args.shutdown and not args.provision:
+            self.parser.error('option --shutdown requires --provision')
+        if args.shutdown and args.allow_password:
+            self.parser.error('option --shutdown conflicts to --allow_password')
+
+        startup_fn = tempfile.mkstemp()[1]
+        with open(startup_fn, 'w') as fobj:
+            data = self._get_ubuntu1404lts_startup(
+                provision=args.provision,
+                shutdown=args.shutdown,
+                homedisk=args.homedisk,
+                allow_password=args.allow_password)
+            fobj.write(data)
+        if args.verbose >= 1 and data:
+            sys.stdout.write("Startup file:\n")
+            with open(startup_fn, 'r') as fobj:
+                lines = fobj.readlines()
+                sys.stdout.write("  " + "  ".join(lines))
+
+        tokens = [_get_trunk(args.project, "compute instances create")]
+        tokens[-1] += " %s" % args.instance_name
+        tokens.append("--machine-type %s" % args.machine_type)
+        if args.zone:
+            tokens.append("--zone %s" % args.zone)
+        tokens.append("--boot-disk-size %s" % args.boot_disk_size)
+        tokens.append("--scopes storage-ro")
+        tokens.append("--metadata-from-file startup-script=%s" % startup_fn)
+        if args.provision:
+            image = "ubuntu-os-cloud/ubuntu-1404-lts"
+        else:
+            image = args.image
+        image_project, image_family = image.split("/")
+        tokens.append("--image-project %s" % image_project)
+        tokens.append("--image-family %s" % image_family)
+        if args.homedisk:
+            tokens.append("--disk=device-name=sdb,mode=rw,name=%s"
+                          % args.homedisk)
+
+        _call_command(tokens, args.verbose, args.dry_run)
+
+        os.unlink(startup_fn)
+
+
+class _MakeImage(object):
```

Comment:
This command creates an image from the boot disk created from _CreateInstance command.